lib/webdriveragent.js

```diff
@@ -218,25 +218,28 @@ class WebDriverAgent {
 
     let status = null;
     let lastError = null;
-    await waitForCondition(async () => {
-      try {
-        status = await getStatus();
-        return true;
-      } catch (err) {
-        lastError = err;
-        return false;
-      }
-    }, {
-      waitMs: this.wdaLaunchTimeout,
-      intervalMs: 300,
-    });
+    try {
+      await waitForCondition(async () => {
+        try {
+          status = await getStatus();
+          return true;
+        } catch (err) {
+          lastError = err;
+          return false;
+        }
+      }, {
+        waitMs: timeoutMs,
+        intervalMs: 300,
+      });
+    } catch (err) {
+      throw new Error(`WDA was not ready in ${timeoutMs} ms`);
```

Comment:
this is actually the place where we need to handle lastError and add it to a message if it is set
the below `if (lastError)` condition is never going to be reached

Reply:
yea, correct. actually right